lib/codec_choose.c

```diff
@@ -1,7 +1,40 @@
 #include <stddef.h>
 
-#if __x86_64__ || __i386__
-#include <cpuid.h>
+#if __x86_64__ || __i386__ || _M_X86 || _M_X64
+	#ifdef _MSC_VER
+		#include <intrin.h>
+		#define __cpuid_count(__level, __count, __eax, __ebx, __ecx, __edx) \
+		{\
+			int info[4];\
+			__cpuidex(info, __level, __count);\
+			__eax = info[0];\
+			__ebx = info[1];\
+			__ecx = info[2];\
+			__edx = info[3];\
+		}
+		#define __cpuid(__level, __eax, __ebx, __ecx, __edx)\
+		{\
+			int info[4];\
+			__cpuidex(info, __level, 0);\
+			__eax = info[0];\
+			__ebx = info[1];\
+			__ecx = info[2];\
+			__edx = info[3];\
+		}
+	#else
+		#include <cpuid.h>
+		#if __GNUC__ > 4 || __GNUC__ == 4 && __GNUC_MINOR__ >= 4
+		inline unsigned long  _xgetbv_32(unsigned int index){
+			unsigned int eax, edx;
+			__asm__ __volatile__("xgetbv" : "=a"(eax), "=d"(edx) : "c"(index));
+			return eax;
+
+		}
+		#else
+			#define _xgetbv_32() 0
```

Comment:
Please use `#error "Platform not supported"` (or similar) here instead of failing silently.